<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="What the package does (one paragraph).">
<title>Calculate Hybrid Index and Heterozygosity from SNP data • triangulaR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Calculate Hybrid Index and Heterozygosity from SNP data">
<meta property="og:description" content="What the package does (one paragraph).">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">triangulaR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/explore_triangle_plots.html">Explore Triangle Plots</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/omys-omics/triangulaR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="triangular-">triangulaR <img src="reference/figures/hex_logo.png" align="right" alt="" width="180"><a class="anchor" aria-label="anchor" href="#triangular-"></a>
</h1></div>
<!-- badges: start -->
<!-- badges: end -->
<p><em>triangulaR</em> is a package for identifying ancestry-informative markers (AIMS), calculating hybrid indices, interclass heterozygosity, and building triangle plots. Check out the articles at the <a href="https://omys-omics.github.io/triangulaR/articles/explore_triangle_plots.html">package website</a> for background on triangle plots, or stay here for a quick start on how to use the package.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of <em>triangulaR</em> from <a href="https://github.com/omys-omics/triangulaR" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"omys-omics/triangulaR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>If you publish any work that uses <em>triangulaR</em>, please consider citing the following paper (currently available as a preprint): <a href="https://www.biorxiv.org/content/10.1101/2024.03.28.587167v1" class="external-link uri">https://www.biorxiv.org/content/10.1101/2024.03.28.587167v1</a></p>
</div>
<div class="section level2">
<h2 id="how-to-use-this-package">How to use this package:<a class="anchor" aria-label="anchor" href="#how-to-use-this-package"></a>
</h2>
<p><em>triangulaR</em> builds on the functionality of <a href="https://knausb.github.io/vcfR_documentation/" class="external-link">vcfR</a> to analyze SNP data in R. The first step is to read in the data from a vcf file.</p>
<div class="section level4">
<h4 id="step-1-read-in-data">Step 1: Read in data<a class="anchor" aria-label="anchor" href="#step-1-read-in-data"></a>
</h4>
<p>It is expected that the data have already filtered for quality (e.g. setting genotype quality and depth thresholds, removing individuals and sites with high missing data, etc.). For help with this step, see <a href="https://github.com/DevonDeRaad/SNPfiltR" class="external-link">SNPfiltR</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/omys-omics/triangulaR" class="external-link">triangulaR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: ggplot2</span></span>
<span><span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span></span>
<span><span class="co">#&gt; This is triangulaR v.0.0.0.9000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           /\</span></span>
<span><span class="co">#&gt;          /  \</span></span>
<span><span class="co">#&gt;         /    \</span></span>
<span><span class="co">#&gt;        /______\</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage information is available at: https://github.com/omys-omics/triangulaR/ </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Please cite the following if you use triangulaR in a publication:</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/knausb/vcfR" class="external-link">vcfR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    *****       ***   vcfR   ***       *****</span></span>
<span><span class="co">#&gt;    This is vcfR 1.14.0 </span></span>
<span><span class="co">#&gt;      browseVignettes('vcfR') # Documentation</span></span>
<span><span class="co">#&gt;      citation('vcfR') # Citation</span></span>
<span><span class="co">#&gt;    *****       *****      *****       *****</span></span>
<span></span>
<span><span class="co"># Read in data</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"../SecondaryContact/gen.19000.vcf"</span>, verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">data</span></span>
<span><span class="co">#&gt; ***** Object of Class vcfR *****</span></span>
<span><span class="co">#&gt; 420 samples</span></span>
<span><span class="co">#&gt; 1 CHROMs</span></span>
<span><span class="co">#&gt; 1,788 variants</span></span>
<span><span class="co">#&gt; Object size: 6.2 Mb</span></span>
<span><span class="co">#&gt; 0 percent missing data</span></span>
<span><span class="co">#&gt; *****        *****         *****</span></span>
<span></span>
<span><span class="co"># Or, use example vcfR object</span></span>
<span><span class="va">example.vcfR</span></span>
<span><span class="co">#&gt; ***** Object of Class vcfR *****</span></span>
<span><span class="co">#&gt; 120 samples</span></span>
<span><span class="co">#&gt; 1 CHROMs</span></span>
<span><span class="co">#&gt; 6,177 variants</span></span>
<span><span class="co">#&gt; Object size: 7.1 Mb</span></span>
<span><span class="co">#&gt; 0 percent missing data</span></span>
<span><span class="co">#&gt; *****        *****         *****</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="step-2-make-a-popmap">Step 2: Make a popmap<a class="anchor" aria-label="anchor" href="#step-2-make-a-popmap"></a>
</h4>
<p>A popmap is a data.frame with two columns labeled “id” and “pop”. Each name of each sample in the vcfR object should be included in the “id” column. All individuals in the vcfR object should be included in the popmap, and vice versa. Each individual needs to be assigned to a population. Every individual must be assigned to a population, and there can be any number of populations. IDs and pops should be character strings.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Here is an example of what a popmap should look like</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/vcfR-method.html" class="external-link">head</a></span><span class="op">(</span><span class="va">example.popmap</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;     id pop</span></span>
<span><span class="co">#&gt; 1  i55  P1</span></span>
<span><span class="co">#&gt; 2 i159  P1</span></span>
<span><span class="co">#&gt; 3 i245  P1</span></span>
<span><span class="co">#&gt; 4 i246  P1</span></span>
<span><span class="co">#&gt; 5 i264  P1</span></span>
<span><span class="co">#&gt; 6 i526  P1</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="va">example.popmap</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;         id pop</span></span>
<span><span class="co">#&gt; 115 bc2.15 BC2</span></span>
<span><span class="co">#&gt; 116 bc2.16 BC2</span></span>
<span><span class="co">#&gt; 117 bc2.17 BC2</span></span>
<span><span class="co">#&gt; 118 bc2.18 BC2</span></span>
<span><span class="co">#&gt; 119 bc2.19 BC2</span></span>
<span><span class="co">#&gt; 120 bc2.20 BC2</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="step-3-choose-sites-above-an-allele-frequency-difference-threshold">Step 3: Choose sites above an allele frequency difference threshold<a class="anchor" aria-label="anchor" href="#step-3-choose-sites-above-an-allele-frequency-difference-threshold"></a>
</h4>
<p>Theoretically, the hybrid index of an individual represents the proportion of ancestry received from each parental population. In practice, one way to calculate hybrid indices is by identifying loci with allele frequency differences above a chosen threshold in the parental populations and scoring individuals by allele counts at those loci. There is a balance between using a small amount of highly diagnostic site (e.g. fixed differences) and a large amount of less diagnostic sites. I recommend trying difference values for the allele frequency difference threshold to see how this value affects results.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a new vcfR object composed only of sites above the given allele frequency difference threshold</span></span>
<span><span class="va">example.vcfR.diff</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/alleleFreqDiff.html">alleleFreqDiff</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="va">example.vcfR</span>, pm <span class="op">=</span> <span class="va">example.popmap</span>, p1 <span class="op">=</span> <span class="st">"P1"</span>, p2 <span class="op">=</span> <span class="st">"P2"</span>, difference <span class="op">=</span> <span class="fl">0.9</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "50 sites passed allele frequency difference threshold"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="step-4-calculate-hybrid-index-and-heterozygosity-for-each-sample">Step 4: Calculate hybrid index and heterozygosity for each sample<a class="anchor" aria-label="anchor" href="#step-4-calculate-hybrid-index-and-heterozygosity-for-each-sample"></a>
</h4>
<p>Once sites above the allele frequency difference threshold have been identified, hybrid index and heterozygosity for each sample can be calculated.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate hybrid index and heterozygosity for each sample. Values are returned in a data.frame</span></span>
<span><span class="va">hi.het</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/hybridIndex.html">hybridIndex</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="va">example.vcfR.diff</span>, pm <span class="op">=</span> <span class="va">example.popmap</span>, p1 <span class="op">=</span> <span class="st">"P1"</span>, p2 <span class="op">=</span> <span class="st">"P2"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "calculating hybrid indices and heterozygosities based on 50 sites"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="step-5-visualize-results-as-a-triangle-plot">Step 5: Visualize results as a triangle plot<a class="anchor" aria-label="anchor" href="#step-5-visualize-results-as-a-triangle-plot"></a>
</h4>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Generate colors (or leave blank to use default)</span></span>
<span><span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#af8dc3"</span>, <span class="st">"#7fbf7b"</span>, <span class="st">"#bababa"</span>, <span class="st">"#878787"</span>, <span class="st">"#762a83"</span>, <span class="st">"#1b7837"</span><span class="op">)</span></span>
<span><span class="co"># View triangle plot</span></span>
<span><span class="fu"><a href="reference/triangle.plot.html">triangle.plot</a></span><span class="op">(</span><span class="va">hi.het</span>, colors <span class="op">=</span> <span class="va">cols</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-5-1.png" width="100%"></p>
</div>
<div class="section level4">
<h4 id="step-6-color-triangle-plot-by-missing-data">Step 6: Color triangle plot by missing data<a class="anchor" aria-label="anchor" href="#step-6-color-triangle-plot-by-missing-data"></a>
</h4>
<p>The data.frame returned by the hybridIndex function also contains the percent of missing data in each individual. View the triangle plot with samples colored by percent missing data to investigate its effect.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># There is no missing data in this dataset, so all samples have the same color</span></span>
<span><span class="fu"><a href="reference/missing.plot.html">missing.plot</a></span><span class="op">(</span><span class="va">hi.het</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-6-1.png" width="100%"></p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/omys-omics/triangulaR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/omys-omics/triangulaR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing triangulaR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Ben Wiens <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-6125-4579" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Ben Wiens.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
