---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# triangulaR

<!-- badges: start -->
<!-- badges: end -->

triangulaR is a package for calculating hybrid indices, heterozygosity, and building triangle plots

## Installation

You can install the development version of triangulaR from [GitHub](https://github.com/omys-omics/triangulaR) with:

``` r
# install.packages("devtools")
devtools::install_github("omys-omics/triangulaR")
```

## Citation

If you publish any work that uses triangulaR, please consider citing the following paper:


## How to use this package:

triangulaR builds on the functionality of [vcfR](https://knausb.github.io/vcfR_documentation/) to analyze SNP data in R. The first step is to read in the data from a vcf file.

#### Step 1: Read in data
It is expected that the data have already filtered for quality (e.g. setting genotype quality and depth thresholds, removing individuals and sites with high missing data, etc.). For help with this step, see [SNPfiltR](https://github.com/DevonDeRaad/SNPfiltR).

```{r example}
library(triangulaR)
library(vcfR)

# Read in data
data <- read.vcfR("../SecondaryContact/gen.19000.vcf", verbose = F)
data

# Or, use example vcfR object
example.vcfR
```

#### Step 2: Make a popmap
A popmap is a data.frame with two columns labeled "id" and "pop". Each name of each sample in the vcfR object should be included in the "id" column. All individuals in the vcfR object should be included in the popmap, and vice versa. Each individual needs to be assigned to a population. Every individual must be assigned to a population, and there can be any number of populations. IDs and pops should be character strings.

```{r}
# Here is an example of what a popmap should look like
print(head(example.popmap))
print(tail(example.popmap))
```

#### Step 3: Choose sites above an allele frequency difference threshold
Theoretically, the hybrid index of an individual represents the proportion of ancestry received from each parental population. In practice, one way to calculate hybrid indices is by identifying loci with allele frequency differences above a chosen threshold in the parental populations and scoring individuals by allele counts at those loci. There is a balance between using a small amount of highly diagnostic site (e.g. fixed differences) and a large amount of less diagnostic sites. I recommend trying difference values for the allele frequency difference threshold to see how this value affects results. 

```{r}
# Create a new vcfR object composed only of sites above the given allele frequency difference threshold
example.vcfR.diff <- alleleFreqDiff(vcfR = example.vcfR, pm = example.popmap, p1 = 0, p2 = 20, difference = 0.6)
```

#### Step 4: Calculate hybrid index and heterozygosity for each sample
Once sites above the allele frequency difference threshold have been identified, hybrid index and heterozygosity for each sample can be calculated. 

```{r}
# Calculate hybrid index and heterozygosity for each sample. Values are returned in a data.frame
hi.het <- hybridIndex(vcfR = example.vcfR.diff, pm = example.popmap, p1 = 0, p2 = 20)
```

#### Step 5: Visualize results as a triangle plot
```{r}
# Generate colors (or leave blank to use default)
cols <- colorRampPalette(colors = c("#313695", "khaki2", "#a50026"))
# View triangle plot
triangle.plot(hi.het, colors = cols(21))
```

#### Step 6: Color triangle plot by missing data
The data.frame returned by the hybridIndex function also contains the percent of missing data in each individual. View the triangle plot with samples colored by percent missing data to investigate its effect.

```{r}
# There is no missing data in this data.set, so all samples have the same color
missing.plot(hi.het)
```


