---
title: "Explore Triangle Plots"
author: "Ben Wiens"
date: "4/4/2024"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# Under Construction

# Introduction

### Punnet Squares

### Hardy-Weinberg Equilibrium

Hardy-Weinberg Equilibirum states that under neutral evolutionary assumptions, the genotype frequencies ($p_{11}$, $p_{12}$, $p_{22}$) in the next generation are given by the allele frequencies in the current generation. Allele frequencies are denoted as $p_1$ and $p_2$, but when there are only two alleles at a locus, $p_2$ can be written as $1-p_1$, yielding the following equations for genotype frequencies.
$$
p_{11} = p_1^2
$$
$$
p_{12} = 2p_1(1-p_1)
$$
$$
p_{22} = (1-p_1)^2
$$
Let's consider the hybrid index and interclass heterozygosity of individuals at sites that are fixed for alternate alelles in the parental populations. When only considering fixed differences, it follows that hybrid index equates to the frequency of p1 alleles across the genome. Similarly, interclass heterozygosity equates to the frequency of p1p2 genotypes across the genomes.

When the parents have the same hybrid index, they can be thought of similarly as a population that is in HWE, and the above formulas can be used to calculate the hybrid index and interclass heterozygosity of offspring. For example,  let's consider a cross between two F1s. Both parents have a hybrid index of 0.5, which means the frequency of the $p_1$ allele is 0.5. Using only this information, we can calculate the genotype frequencies of the offspring (F2s).

$$
p_{11} = 0.5^2 = 0.25
$$
$$
p_{12} = 2(0.5)(1-0.5) = 0.5
$$
$$
p_{22} = (1-0.5)^2 =0.25
$$
From the genotype frequencies of the offspring, we can obtain the frequency of the $p_1$ allele by dividing the frequency of $p_12$ genotype by $2$ (because the  and the $p_1$ allele only occurs once in that genotype) and summing with the frequency of the $p_11$ genotype:
$$
p_1 = 0.5(p_{12}) + p_{11} = 0.5(0.5) + 0.25 = 0.5
$$
Therefore, we obtain the expected hybrid index (0.5, frequency of the $p_1$ allele) and interclass heterozygosity (0.5, frequency of the $p_{12$ genotype) in the F2 generation. 

However, for crosses between parents with different hybrid indices, we must slighly modify the above formulas, because the parents are no longer in HWE. To calculate the genotype frequencies for offspring in such cases, we must take into account the difference in hybrid index between the two parents. This is done by calculating the variance ($σ^2$) of the $p_1$ allele frequency in the two parents:
$$
σ^2 = \frac{Σ (p_{1i} − \overline{p_1})^2} {N}
$$
where $p_{1i}$ is the frequency of the $p_1$ allele in the $ith$ parent and $\overline{p_1}$ is the avereage frequency of the $p_1$ allele in the parents (Hahn, 2018).


Based on the above formulas, we can calculate the hybrid index and interclass heterozygosity of any offspring given only the average allele frequency of p1 alleles in the parents. 


![All offspring through six generations, after all possible matings in the previous generation occur](../man/figures/FigureS1.png)

### Try it out


Example with simulated data:
1. See distribution of allele frequency differences
2. Get names of AIMs
3. Try multiple thresholds and get vcfRs
4. Calculate HI and HET
5. Compare triangle plots







</style>

<div class="shiny-app-frame"> 
<iframe src="https://omys-omics.shinyapps.io/Hybrid_Offspring/" width="1000" height="1000">
</iframe>
</div>



Example with real data:
1. See distribution of allele frequency differences
2. Try multiple thresholds and get vcfRs
3. Calculate HI and HET
4. Compare triangle plots
5. Color by missing data

### Try changing parameters to see how they impact hybrid identification





</style>

<div class="shiny-app-frame"> 
<iframe src="https://omys-omics.shinyapps.io/Simulate_hybrid_parameters/" width="1000" height="1200">
</iframe>
</div>


### References
Hahn, M. W. (2018). Molecular population genetics. Oxford University Press.; Sinauer 
Associates.



