---
title: "Triangle Plot Basics"
author: "Ben Wiens"
date: "4/9/2024"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### Identifying AIMs

In order to build triangle plots, the first thing we need to do is identify ancestry-informative markers (AIMs). These are SNPs that have alleles that are sorted between two distinct parental groups, which means that each allele at a SNP indicates ancestry from one parental group or the other. In this example, we are going to assume that there are only two alleles at each SNP, which is also how empircal datasets are often treated. Consider the two populations shown below. Each population consists of diploid organisms, and six SNPs are shown on the example chromosomes. We can see that population 1 only has red alleles at SNPs 1, 2, 4, and 6, and that there are some blue alleles at SNPs 3 and 5. If we look at population 2, we see that there are only blue alleles at SNPs 1, 2, 3, 4, and 6. So, both populations have both alleles at site 5, and population 1 has both alleles at site 3. Therefore, only sites 1, 2, 4, and 6 are completely informative ancestry. When alleles at a SNP are completely sorted between populations, we call this a fixed difference. Moving forward in this example, we will use SNPs 1, 2, 4, and 6 as AIMs.

![](../man/figures/parental_pops.png)
$$\\[0.05in]$$


### Calculating Hybrid Index and Interclass Heterozygosity


<img src="../man/figures/first_cross.png" align="left" alt="" width="280" /> One thing triangle plots can help do is identify early generation hybrids, such as F1s. An F1 is the offspring of a cross between an individual from parental population 1 and parental population 2. The illustration to the left shows the alleles present at the AIMs we have identified. Because offspring inherit one allele from each parent, we can see that the F1 has a blue allele and a red allele at each AIM. In order to build a triangle plot, we first need to calculate hybrid index and interclass heterozygosity. Hybrid index is calculated by counting the number of alleles inherited from parental population 2, and dividing by the total number of alleles. This gives us a value that can range between 0 and 1. In this case, the F1 has four alleles from parental population 2 (blue), and eight total alleles. This gives a hybrid index of $\frac{4}{8}=0.5$. Interclass heterozygosity is calculated by counting the number of AIMs that have an allele from both parental populations, and dividing by total number of AIMs. In this case there are four AIMs that have an allele from both parental populations, and four total AIMs, meaning the interclass heterozygosity is $\frac{4}{4}=1$.

$$\\[0.01in]$$

<img src="../man/figures/second_cross.png" align="left" alt="" width="280" /> Let's consider another type of cross. The illustration to the left shows the offspring of a cross between two F1s, called an F2. We can calculate hybrid index and interclass heterozygosity for the F2 manually as we did above. In the F2, we see there are four alleles from parental population 2 (blue), so hybrid index is $\frac{4}{8}=0.5$. In contrast to the F1 though, the F2 only has two AIMs with alleles from both parental populations, so interclass heterozygosity is $\frac{2}{4}=0.5$. These are the expected values for an F2, as we will see in the next section.
$$\\[0.01in]$$

### Punnett Squares


### Borders of Triangle Plot
