<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>read.genotypes — read.genotypes • triangulaR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="read.genotypes — read.genotypes"><meta name="description" content="Read in genotype data from a tab-delimited file in matrix format (e.g. structure format) and convert it to a vcfR object"><meta property="og:description" content="Read in genotype data from a tab-delimited file in matrix format (e.g. structure format) and convert it to a vcfR object"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">triangulaR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/triangle_plot_basics.html">Triangle Plot Basics</a></li>
    <li><a class="dropdown-item" href="../articles/explore_triangle_plots.html">Explore Triangle Plots</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/omys-omics/triangulaR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>read.genotypes</h1>
      <small class="dont-index">Source: <a href="https://github.com/omys-omics/triangulaR/blob/master/R/functions.R" class="external-link"><code>R/functions.R</code></a></small>
      <div class="d-none name"><code>read.genotypes.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Read in genotype data from a tab-delimited file in matrix format (e.g. structure format) and convert it to a vcfR object</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">read.genotypes</span><span class="op">(</span></span>
<span>  file <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  data.type <span class="op">=</span> <span class="st">"alleles"</span>,</span>
<span>  missing.value <span class="op">=</span> <span class="st">"N"</span>,</span>
<span>  n.id.cols <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  n.id.rows <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  second.allele <span class="op">=</span> <span class="st">"NA"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-file">file<a class="anchor" aria-label="anchor" href="#arg-file"></a></dt>
<dd><p>Name of the tab-delimited file. Individuals should be in rows and alleles/genotypes should be in columns.</p></dd>


<dt id="arg-data-type">data.type<a class="anchor" aria-label="anchor" href="#arg-data-type"></a></dt>
<dd><p>(character) Either "alleles" or "genotypes". Use alleles if each allele (e.g. A,C,G,T OR 1,2,3,4) is encoded separately in the file, either across rows or columns. Use genotypes if the alleles are encoded as a single genotype (e.g. 0=homozygous ref, 1=heterozygous, 2=homozygous alt). Default is alleles.</p></dd>


<dt id="arg-missing-value">missing.value<a class="anchor" aria-label="anchor" href="#arg-missing-value"></a></dt>
<dd><p>(character) Value used for missing data. Default is "N".</p></dd>


<dt id="arg-n-id-cols">n.id.cols<a class="anchor" aria-label="anchor" href="#arg-n-id-cols"></a></dt>
<dd><p>(numeric) Number of columns before genotypes start. Must be at least 1 with individual IDs.</p></dd>


<dt id="arg-n-id-rows">n.id.rows<a class="anchor" aria-label="anchor" href="#arg-n-id-rows"></a></dt>
<dd><p>(numeric) Number of rows before genotypes start. Usually there is 1 with SNP IDs. If not, artificial IDs will be assigned.</p></dd>


<dt id="arg-second-allele">second.allele<a class="anchor" aria-label="anchor" href="#arg-second-allele"></a></dt>
<dd><p>(numeric) Either "columns" or "rows". Use columns if second allele is in the next column. Use rows if second allele is in the next row. Not required if data are in genotype format.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>vcfR object</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#read.genotypes(file = alleles.str, data.type = "alleles", missing.value = -9, n.id.cols = 2, n.id.rows = 1, second.allele = "rows")</span></span></span>
<span class="r-in"><span><span class="co">#read.genotypes(file = genotypes.txt, data.type = "genotypes", missing.value = "N", n.id.cols = 2, n.id.rows = 1)</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ben Wiens.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

