<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Explore Triangle Plots • triangulaR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Explore Triangle Plots">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">triangulaR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/triangle_plot_basics.html">Triangle Plot Basics</a></li>
    <li><a class="dropdown-item" href="../articles/explore_triangle_plots.html">Explore Triangle Plots</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/omys-omics/triangulaR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Explore Triangle Plots</h1>
                        <h4 data-toc-skip class="author">Ben Wiens</h4>
            
            <h4 data-toc-skip class="date">4/4/2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/omys-omics/triangulaR/blob/master/vignettes/explore_triangle_plots.Rmd" class="external-link"><code>vignettes/explore_triangle_plots.Rmd</code></a></small>
      <div class="d-none name"><code>explore_triangle_plots.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p><img src="../reference/figures/labeled_tri_plot.png" align="right" alt="" width="400">
Triangle plots are a tool for visualizing genetic variation in hybrid
zones and identifying hybrid classes. Triangle plots visualize hybrid
index on the x-axis and interclass heterozygosity on the y-axis, and get
their name because the possible space on the plot forms a triangle.
Hybrid index refers to the ancestry proportions of an individual, and
interclass heterozygosity refers to the genomic sites with an allele
from both parental groups. A number of discrete hybrid classes can be
identified by their expected combination of hybrid index and interclass
heterozygosity. The placement of parentals (P1, P2), first generation
hybrids (F1s), later generation hybrids (F2s and later), and backcrosses
(BC) on a triangle plot is shown to the right For more background on
triangle plots, see the <a href="https://omys-omics.github.io/triangulaR/articles/triangle_plot_basics.html">Triangle
Plot Basics</a> page.</p>
</div>
<div class="section level3">
<h3 id="hardy-weinberg-equilibrium">Hardy-Weinberg Equilibrium<a class="anchor" aria-label="anchor" href="#hardy-weinberg-equilibrium"></a>
</h3>
<p>Hardy-Weinberg Equilibirum states that under neutral evolutionary
assumptions, the genotype frequencies
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>11</mn></msub><annotation encoding="application/x-tex">p_{11}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>12</mn></msub><annotation encoding="application/x-tex">p_{12}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>22</mn></msub><annotation encoding="application/x-tex">p_{22}</annotation></semantics></math>)
in the next generation are given by the allele frequencies in the
current generation. Allele frequencies are denoted as
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>1</mn></msub><annotation encoding="application/x-tex">p_1</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>2</mn></msub><annotation encoding="application/x-tex">p_2</annotation></semantics></math>,
but when there are only two alleles at a locus,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>2</mn></msub><annotation encoding="application/x-tex">p_2</annotation></semantics></math>
can be written as
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>1</mn><mo>−</mo><msub><mi>p</mi><mn>1</mn></msub></mrow><annotation encoding="application/x-tex">1-p_1</annotation></semantics></math>,
yielding the following equations for genotype frequencies.
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>11</mn></msub><mo>=</mo><msubsup><mi>p</mi><mn>1</mn><mn>2</mn></msubsup></mrow><annotation encoding="application/x-tex">
p_{11} = p_1^2
</annotation></semantics></math><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>12</mn></msub><mo>=</mo><mn>2</mn><msub><mi>p</mi><mn>1</mn></msub><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><msub><mi>p</mi><mn>1</mn></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">
p_{12} = 2p_1(1-p_1)
</annotation></semantics></math><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>22</mn></msub><mo>=</mo><msup><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><msub><mi>p</mi><mn>1</mn></msub><mo stretchy="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">
p_{22} = (1-p_1)^2
</annotation></semantics></math> Let’s consider the hybrid index and
interclass heterozygosity of individuals at sites that are fixed for
alternate alelles in the parental populations. When only considering
fixed differences, it follows that hybrid index equates to the frequency
of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>1</mn></msub><annotation encoding="application/x-tex">p_1</annotation></semantics></math>
alleles across the genome. Similarly, interclass heterozygosity equates
to the frequency of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>12</mn></msub><annotation encoding="application/x-tex">p_{12}</annotation></semantics></math>
genotypes across the genome.</p>
<p>When the parents have the same hybrid index, they can be thought of
similarly as a population that is in HWE, and the above formulas can be
used to calculate the hybrid index and interclass heterozygosity of
offspring. For example, let’s consider a cross between two F1s. Both
parents have a hybrid index of 0.5, which means the frequency of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>1</mn></msub><annotation encoding="application/x-tex">p_1</annotation></semantics></math>
allele across the genome is 0.5. Using only this information, we can
calculate the genome-wide genotype frequencies of the offspring
(F2s).</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>11</mn></msub><mo>=</mo><msup><mn>0.5</mn><mn>2</mn></msup><mo>=</mo><mn>0.25</mn></mrow><annotation encoding="application/x-tex">
p_{11} = 0.5^2 = 0.25
</annotation></semantics></math><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>12</mn></msub><mo>=</mo><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mn>0.5</mn><mo stretchy="true" form="postfix">)</mo></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mn>0.5</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><mn>0.5</mn></mrow><annotation encoding="application/x-tex">
p_{12} = 2(0.5)(1-0.5) = 0.5
</annotation></semantics></math><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>22</mn></msub><mo>=</mo><msup><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mn>0.5</mn><mo stretchy="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mo>=</mo><mn>0.25</mn></mrow><annotation encoding="application/x-tex">
p_{22} = (1-0.5)^2 =0.25
</annotation></semantics></math> From the genotype frequencies of the
offspring, we can obtain the frequency of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>1</mn></msub><annotation encoding="application/x-tex">p_1</annotation></semantics></math>
allele by dividing the frequency of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>12</mn></msub><annotation encoding="application/x-tex">p_{12}</annotation></semantics></math>
genotype by
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>2</mn><annotation encoding="application/x-tex">2</annotation></semantics></math>
(because the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>1</mn></msub><annotation encoding="application/x-tex">p_1</annotation></semantics></math>
allele only occurs once in that genotype) and summing with the frequency
of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>11</mn></msub><annotation encoding="application/x-tex">p_{11}</annotation></semantics></math>
genotype:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>1</mn></msub><mo>=</mo><mn>0.5</mn><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>p</mi><mn>12</mn></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><msub><mi>p</mi><mn>11</mn></msub><mo>=</mo><mn>0.5</mn><mrow><mo stretchy="true" form="prefix">(</mo><mn>0.5</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><mn>0.25</mn><mo>=</mo><mn>0.5</mn></mrow><annotation encoding="application/x-tex">
p_1 = 0.5(p_{12}) + p_{11} = 0.5(0.5) + 0.25 = 0.5
</annotation></semantics></math> Therefore, we obtain the expected
hybrid index (0.5, frequency of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>1</mn></msub><annotation encoding="application/x-tex">p_1</annotation></semantics></math>
allele) and interclass heterozygosity (0.5, frequency of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>12</mn></msub><annotation encoding="application/x-tex">p_{12}</annotation></semantics></math>
genotype) in the F2 generation.</p>
<p>However, for crosses between parents with different hybrid indices,
we must slighly modify the above formulas, because the parents are no
longer in HWE. To calculate the genotype frequencies for offspring in
such cases, we must take into account the difference in hybrid index
between the two parents. This is done by calculating the variance
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>σ</mi><mn>2</mn></msup><annotation encoding="application/x-tex">σ^2</annotation></semantics></math>)
of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>1</mn></msub><annotation encoding="application/x-tex">p_1</annotation></semantics></math>
allele frequency in the two parents:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mi>σ</mi><mn>2</mn></msup><mo>=</mo><mfrac><mrow><mi>Σ</mi><msup><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>p</mi><mrow><mn>1</mn><mi>i</mi></mrow></msub><mo>−</mo><mover><msub><mi>p</mi><mn>1</mn></msub><mo accent="true">¯</mo></mover><mo stretchy="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mrow><mi>N</mi></mfrac></mrow><annotation encoding="application/x-tex">
σ^2 = \frac{Σ (p_{1i} − \overline{p_1})^2} {N}
</annotation></semantics></math> where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mrow><mn>1</mn><mi>i</mi></mrow></msub><annotation encoding="application/x-tex">p_{1i}</annotation></semantics></math>
is the frequency of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>1</mn></msub><annotation encoding="application/x-tex">p_1</annotation></semantics></math>
allele in the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>i</mi><mi>t</mi><mi>h</mi></mrow><annotation encoding="application/x-tex">ith</annotation></semantics></math>
parent and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mover><msub><mi>p</mi><mn>1</mn></msub><mo accent="true">¯</mo></mover><annotation encoding="application/x-tex">\overline{p_1}</annotation></semantics></math>
is the average frequency of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>1</mn></msub><annotation encoding="application/x-tex">p_1</annotation></semantics></math>
allele in the parents (Hahn, 2018). Genotype frequencies of offspring
are then calculated as:
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>11</mn></msub><mo>=</mo><msup><mover><msub><mi>p</mi><mn>1</mn></msub><mo accent="true">¯</mo></mover><mn>2</mn></msup><mo>−</mo><msup><mi>σ</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">
p_{11} = \overline{p_1}^2 - σ^2 
</annotation></semantics></math><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>12</mn></msub><mo>=</mo><mn>2</mn><mover><msub><mi>p</mi><mn>1</mn></msub><mo accent="true">¯</mo></mover><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mover><msub><mi>p</mi><mn>1</mn></msub><mo accent="true">¯</mo></mover><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><mn>2</mn><msup><mi>σ</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">
p_{12} = 2\overline{p_1}(1-\overline{p_1}) + 2σ^2 
</annotation></semantics></math><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>22</mn></msub><mo>=</mo><msup><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mover><msub><mi>p</mi><mn>1</mn></msub><mo accent="true">¯</mo></mover><mo stretchy="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mo>−</mo><msup><mi>σ</mi><mn>2</mn></msup></mrow><annotation encoding="application/x-tex">
p_{22} = (1-\overline{p_1})^2 - σ^2 
</annotation></semantics></math></p>
<p>Based on the above formulas, we can calculate the hybrid index and
interclass heterozygosity of any offspring given only the average allele
frequency of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>1</mn></msub><annotation encoding="application/x-tex">p_1</annotation></semantics></math>
alleles in the parents. Try it out for yourself by adjusting the hybrid
index and interclass heterozygosity of the parents, and seeing where the
hybrid offspring would occur on a triangle plot:</p>

<div class="shiny-app-frame">
<iframe src="https://omys-omics.shinyapps.io/Hybrid_Offspring/" width="1000" height="550">
</iframe>
</div>
<p>The dashed curve on the above triangle plot denotes the expected
hybrid index and interclass heterozygosity under HWE. As you may have
noticed, hybrid offspring will never occur below the dashed curve on the
triangle plot. Why is that? We can understand this by considering two
cases of crosses. In the first case, the parents have the same hybrid
index. When this is the case, offspring will occur on the dashed line by
definition, because the parents are in HWE. In the second case, the
parents have different hybrid indices. In this case, the hybrid index of
offspring will be the same as for a cross between parents of the average
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>1</mn></msub><annotation encoding="application/x-tex">p_1</annotation></semantics></math>
allele frequency, because the genotype frequencies
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>11</mn></msub><annotation encoding="application/x-tex">p_{11}</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>22</mn></msub><annotation encoding="application/x-tex">p_{22}</annotation></semantics></math>
will decrease by the same increment
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>σ</mi><mn>2</mn></msup><annotation encoding="application/x-tex">σ^2</annotation></semantics></math>),
and changes in the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>12</mn></msub><annotation encoding="application/x-tex">p_{12}</annotation></semantics></math>
genotype frequency don’t change the hybrid index. However, the
interclass heterozygosity will be higher than for a cross between
parents of the average
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>1</mn></msub><annotation encoding="application/x-tex">p_1</annotation></semantics></math>
allele frequency, because the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>12</mn></msub><annotation encoding="application/x-tex">p_{12}</annotation></semantics></math>
genotype frequency will increase by
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>σ</mi><mn>2</mn></msup><annotation encoding="application/x-tex">σ^2</annotation></semantics></math>.
Thus, hybrid offspring will always occur on or above the dashed curve.
Of course, this is only if the assumptions of HWE hold. When any of the
assumptions of HWE are violated, such as if there is selection, drift,
or inbreeding, it would be possible for individuals to occur below the
dashed curve.</p>
<p>We can arrive at the same conclusions about the possible space on a
triangle plot under HWE by simulating all possible matings through six
generations, starting with only parental individuals. After the first
generation, only parentals and F1s are present, shown in the first panel
below. Each subsequent panel shows the hybrid offspring created by
crossing all individuals present in the previous generation with every
other individual.</p>
<div class="float">
<img src="../reference/figures/FigureS1.png" alt="All offspring through six generations, after all possible matings in the previous generation occur"><div class="figcaption">All offspring through six generations, after all
possible matings in the previous generation occur</div>
</div>
<p><span class="math display">$$\\[0.15in]$$</span></p>
</div>
<div class="section level3">
<h3 id="triangle-plots-in-practice">Triangle plots in practice<a class="anchor" aria-label="anchor" href="#triangle-plots-in-practice"></a>
</h3>
<p>SNP datasets generated through genomic sequencing (e.g. WGS, RADseq,
target capture) of individuals from hybrid zones can be used to build
triangle plots with the R package <a href="https://github.com/omys-omics/triangulaR" class="external-link"><em>triangulaR</em></a>.
The method employeed here depends on the sampling of parental
populations in order to identify ancestry-informative markers (AIMs).
SNPs with a high allele frequency difference between the parental
populations can be used as AIMs to calculate hybrid index and interclass
heterozygosity.</p>
<p>When analyzing empirical genetic datasets, there are a number of
factors that will influence the accuracy and precision of estimated
hybrid indices and interclass heterozygosity (Wiens &amp; Colella,
2024). Some factors are out of our control, such as how divergent the
parental populations are. Other factors we can control, such as how many
parental individuals are sampled, how much of the genome is sequenced,
and the allele frequency difference threshold used for identifying AIMs.
The goal is to identify a large dataset of SNPs with high allele
frequency differences between the parental populations. When divergence
between parental populations is high, there will be many AIMs.
Sequencing more of the genome (e.g. WGS instead of RADseq) will also
provide more AIMs. Including more samples from the parental populations
will increase the accuracy of estimated allele frequency differences,
and thereby increase the accuracy of hybrid index and interclasss
heterozygosity estimates. Lower allele frequency difference thresholds
will increase the size of the AIM dataset, and allow comparisons to be
made between the parental populations.</p>
<p>These factors all interact to influence the placement of individuals
on a triangle plot. In the interactive plot below, 20 individuals from
four hybrid classes (F1, F2, backcrosses in both directions) are
simulated and plotted based on the imput parameters. Try playing with
the parameters to see how they impact identification of hybrid
classes.</p>

<div class="shiny-app-frame">
<iframe src="https://omys-omics.shinyapps.io/Simulate_hybrid_parameters/" width="1000" height="1000">
</iframe>
</div>
</div>
<div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<p>Hahn, M.W. (2018). <em>Molecular population genetics</em>. Oxford
University Press. Sinauer Associates.<br>
Wiens, B.J., &amp; Colella, J.P. (2024). <em>triangulaR: an R package
for identifying AIMs</em> <em>and building triangle plots using SNP data
from hybrid zones</em>. bioRxiv, 2024.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ben Wiens.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
