<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="triangulaR">
<title>Explore Triangle Plots • triangulaR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Explore Triangle Plots">
<meta property="og:description" content="triangulaR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">triangulaR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/explore_triangle_plots.html">Explore Triangle Plots</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/omys-omics/triangulaR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Explore Triangle Plots</h1>
                        <h4 data-toc-skip class="author">Ben Wiens</h4>
            
            <h4 data-toc-skip class="date">4/4/2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/omys-omics/triangulaR/blob/HEAD/vignettes/explore_triangle_plots.Rmd" class="external-link"><code>vignettes/explore_triangle_plots.Rmd</code></a></small>
      <div class="d-none name"><code>explore_triangle_plots.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="under-construction">Under Construction<a class="anchor" aria-label="anchor" href="#under-construction"></a>
</h2>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Triangle plots are a tool for visualizing genetic variation in hybrid
zones and identifying hybrid classes. Triangle plots visualize hybrid
index on the x-axis and interclass heterozygosity on the y-axis, and get
their name because the possible space on the plot forms a triangle.
Hybrid index refers to the ancestry proportions of an individual, and
interclass heterozygosity refers to the genomic sites with an allele
from both parental groups. A number of discrete hybrid classes can be
identified by their expected combination of hybrid index and interclass
heterozygosity. The placement of parentals (P1, P2), first generation
hybrids (F1s), later generation hybrids (F2s and later), and backcrosses
(BC) on a triangle plot is shown below.</p>
<p><img src="../reference/figures/labeled_tri_plot.png"></p>
<div class="section level4">
<h4 id="hardy-weinberg-equilibrium">Hardy-Weinberg Equilibrium<a class="anchor" aria-label="anchor" href="#hardy-weinberg-equilibrium"></a>
</h4>
<p>Hardy-Weinberg Equilibirum states that under neutral evolutionary
assumptions, the genotype frequencies (<span class="math inline">\(p_{11}\)</span>, <span class="math inline">\(p_{12}\)</span>, <span class="math inline">\(p_{22}\)</span>) in the next generation are given
by the allele frequencies in the current generation. Allele frequencies
are denoted as <span class="math inline">\(p_1\)</span> and <span class="math inline">\(p_2\)</span>, but when there are only two alleles
at a locus, <span class="math inline">\(p_2\)</span> can be written as
<span class="math inline">\(1-p_1\)</span>, yielding the following
equations for genotype frequencies. <span class="math display">\[
p_{11} = p_1^2
\]</span> <span class="math display">\[
p_{12} = 2p_1(1-p_1)
\]</span> <span class="math display">\[
p_{22} = (1-p_1)^2
\]</span> Let’s consider the hybrid index and interclass heterozygosity
of individuals at sites that are fixed for alternate alelles in the
parental populations. When only considering fixed differences, it
follows that hybrid index equates to the frequency of <span class="math inline">\(p_1\)</span> alleles across the genome. Similarly,
interclass heterozygosity equates to the frequency of <span class="math inline">\(p_{12}\)</span> genotypes across the genome.</p>
<p>When the parents have the same hybrid index, they can be thought of
similarly as a population that is in HWE, and the above formulas can be
used to calculate the hybrid index and interclass heterozygosity of
offspring. For example, let’s consider a cross between two F1s. Both
parents have a hybrid index of 0.5, which means the frequency of the
<span class="math inline">\(p_1\)</span> allele across the genome is
0.5. Using only this information, we can calculate the genome-wide
genotype frequencies of the offspring (F2s).</p>
<p><span class="math display">\[
p_{11} = 0.5^2 = 0.25
\]</span> <span class="math display">\[
p_{12} = 2(0.5)(1-0.5) = 0.5
\]</span> <span class="math display">\[
p_{22} = (1-0.5)^2 =0.25
\]</span> From the genotype frequencies of the offspring, we can obtain
the frequency of the <span class="math inline">\(p_1\)</span> allele by
dividing the frequency of <span class="math inline">\(p_{12}\)</span>
genotype by <span class="math inline">\(2\)</span> (because the <span class="math inline">\(p_1\)</span> allele only occurs once in that
genotype) and summing with the frequency of the <span class="math inline">\(p_11\)</span> genotype: <span class="math display">\[
p_1 = 0.5(p_{12}) + p_{11} = 0.5(0.5) + 0.25 = 0.5
\]</span> Therefore, we obtain the expected hybrid index (0.5, frequency
of the <span class="math inline">\(p_1\)</span> allele) and interclass
heterozygosity (0.5, frequency of the <span class="math inline">\(p_{12}\)</span> genotype) in the F2
generation.</p>
<p>However, for crosses between parents with different hybrid indices,
we must slighly modify the above formulas, because the parents are no
longer in HWE. To calculate the genotype frequencies for offspring in
such cases, we must take into account the difference in hybrid index
between the two parents. This is done by calculating the variance (<span class="math inline">\(σ^2\)</span>) of the <span class="math inline">\(p_1\)</span> allele frequency in the two parents:
<span class="math display">\[
σ^2 = \frac{Σ (p_{1i} − \overline{p_1})^2} {N}
\]</span> where <span class="math inline">\(p_{1i}\)</span> is the
frequency of the <span class="math inline">\(p_1\)</span> allele in the
<span class="math inline">\(ith\)</span> parent and <span class="math inline">\(\overline{p_1}\)</span> is the average frequency
of the <span class="math inline">\(p_1\)</span> allele in the parents
(Hahn, 2018). Genotype frequencies of offspring are then calculated as:
<span class="math display">\[
p_{11} = \overline{p_1}^2 - σ^2
\]</span> <span class="math display">\[
p_{12} = 2\overline{p_1}(1-\overline{p_1}) + 2σ^2
\]</span> <span class="math display">\[
p_{22} = (1-\overline{p_1})^2 - σ^2
\]</span></p>
<p>Based on the above formulas, we can calculate the hybrid index and
interclass heterozygosity of any offspring given only the average allele
frequency of <span class="math inline">\(p_1\)</span> alleles in the
parents. Try it out for yourself by adjusting the hybrid index and
interclass heterozygosity of the parents, and seeing where the hybrid
offspring would occur on a triangle plot:</p>

<div class="shiny-app-frame">
<iframe src="https://omys-omics.shinyapps.io/Hybrid_Offspring/" width="1000" height="550">
</iframe>
</div>
<p>The dashed curve on the above triangle plot denotes the expected
hybrid index and interclass heterozygosity under HWE. As you may have
noticed, hybrid offspring will never occur below the dashed curve on the
triangle plot. Why is that? We can understand this by considering two
cases of crosses. In the first case, the parents have the same hybrid
index. When this is the case, offspring will occur on the dashed line by
definition, because the parents are in HWE. In the second case, the
parents have different hybrid indices. In this case, the hybrid index of
offspring will be the same as for a cross between parents of the average
<span class="math inline">\(p_1\)</span> allele frequency, because the
genotype frequencies <span class="math inline">\(p_{11}\)</span> and
<span class="math inline">\(p_{22}\)</span> will decrease by the same
increment (<span class="math inline">\(σ^2\)</span>), and changes in the
<span class="math inline">\(p_{12}\)</span> genotype frequency don’t
change the hybrid index. However, the interclass heterozygosity will be
higher than for a cross between parents of the average <span class="math inline">\(p_1\)</span> allele frequency, because the <span class="math inline">\(p_{12}\)</span> genotype frequency will increase
by <span class="math inline">\(σ^2\)</span>. Thus, hybrid offspring will
always occur on or above the dashed curve. Of course, this is only if
the assumptions of HWE hold. When any of the assumptions of HWE are
violated, such as if there is selection, drift, or inbreeding, it would
be possible for individuals to occur below the dashed curve.</p>
<p>We can arrive at the same conclusions about the possible space on a
triangle plot under HWE by simulating all possible matings through six
generations, starting with only parental individuals. After the first
generation, only parentals and F1s are present, shown in the first panel
below. Each subsequent panel shows the hybrid offspring created by
crossing all individuals present in the previous generation with every
other individual.</p>
<div class="float">
<img src="../reference/figures/FigureS1.png" alt="All offspring through six generations, after all possible matings in the previous generation occur"><div class="figcaption">All offspring through six generations, after all
possible matings in the previous generation occur</div>
</div>
</div>
<div class="section level4">
<h4 id="triangle-plots-in-practice">Triangle plots in practice<a class="anchor" aria-label="anchor" href="#triangle-plots-in-practice"></a>
</h4>
<p>SNP datasets generated through genomic sequencing (e.g. WGS, RADseq,
target capture) of individuals from hybrid zones can be used to build
triangle plots with the R package <a href="https://github.com/omys-omics/triangulaR" class="external-link"><em>triangulaR</em></a>.
The method employeed here depends on the sampling of parental
populations in order to identify ancestry-informative markers (AIMs).
SNPs with a high allele frequency difference between the parental
populations can be used as AIMs to calculate hybrid index and interclass
heterozygosity.</p>
<p>When analyzing empirical genetic datasets, there are a number of
factors that will influence the accuracy and precision of estimated
hybrid indices and interclass heterozygosity (Wiens &amp; Colella,
2024). Some factors are out of our control, such as how divergent the
parental populations are. Other factors we can control, such as how many
parental individuals are sampled, how much of the genome is sequenced,
and the allele frequency difference threshold used for identifying AIMs.
The goal is to identify a large dataset of SNPs with high allele
frequency differences between the parental populations. When divergence
between parental populations is high, there will be many AIMs.
Sequencing more of the genome (e.g. WGS instead of RADseq) will also
provide more AIMs. Including more samples from the parental populations
will increase the accuracy of estimated allele frequency differences,
and thereby increase the accuracy of hybrid index and interclasss
heterozygosity estimates. Lower allele frequency difference thresholds
will increase the size of the AIM dataset, and allow comparisons to be
made between the parental populations.</p>
<p>These factors all interact to influence the placement of individuals
on a triangle plot. In the interactive plot below, 20 individuals from
four hybrid classes (F1, F2, backcrosses in both directions) are
simulated and plotted based on the imput parameters. Try playing with
the parameters to see how they impact identification of hybrid
classes.</p>

<div class="shiny-app-frame">
<iframe src="https://omys-omics.shinyapps.io/Simulate_hybrid_parameters/" width="1000" height="1000">
</iframe>
</div>
</div>
<div class="section level4">
<h4 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h4>
<p>Hahn, M.W. (2018). <em>Molecular population genetics</em>. Oxford
University Press. Sinauer Associates.<br>
Wiens, B.J., &amp; Colella, J.P. (2024). <em>triangulaR: an R package
for identifying AIMs</em> <em>and building triangle plots using SNP data
from hybrid zones</em>. bioRxiv, 2024.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Ben Wiens.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
