<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Triangle Plot Basics • triangulaR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Triangle Plot Basics">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">triangulaR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/triangle_plot_basics.html">Triangle Plot Basics</a></li>
    <li><a class="dropdown-item" href="../articles/explore_triangle_plots.html">Explore Triangle Plots</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/omys-omics/triangulaR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Triangle Plot Basics</h1>
                        <h4 data-toc-skip class="author">Ben Wiens</h4>
            
            <h4 data-toc-skip class="date">4/9/2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/omys-omics/triangulaR/blob/master/vignettes/triangle_plot_basics.Rmd" class="external-link"><code>vignettes/triangle_plot_basics.Rmd</code></a></small>
      <div class="d-none name"><code>triangle_plot_basics.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="identifying-aims">Identifying AIMs<a class="anchor" aria-label="anchor" href="#identifying-aims"></a>
</h3>
<p>In order to build triangle plots, the first thing we need to do is
identify ancestry-informative markers (AIMs). These are SNPs that have
alleles that are sorted between two distinct parental groups, which
means that each allele at a SNP indicates ancestry from one parental
group or the other. In this example, we are going to assume that there
are only two alleles at each SNP, which is also how empircal datasets
are often treated. Consider the two populations shown below. Each
population consists of diploid organisms, and six SNPs are shown on the
example chromosomes. We can see that population 1 only has red alleles
at SNPs 1, 2, 4, and 6, and that there are some blue alleles at SNPs 3
and 5. If we look at population 2, we see that there are only blue
alleles at SNPs 1, 2, 3, 4, and 6. So, both populations have both
alleles at site 5, and population 1 has both alleles at site 3.
Therefore, only sites 1, 2, 4, and 6 are completely informative
ancestry. When alleles at a SNP are completely sorted between
populations, we call this a fixed difference. Moving forward in this
example, we will use SNPs 1, 2, 4, and 6 as AIMs.</p>
<p><img src="../reference/figures/parental_pops.png"><span class="math display">$$\\[0.05in]$$</span></p>
</div>
<div class="section level3">
<h3 id="calculating-hybrid-index-and-interclass-heterozygosity">Calculating Hybrid Index and Interclass Heterozygosity<a class="anchor" aria-label="anchor" href="#calculating-hybrid-index-and-interclass-heterozygosity"></a>
</h3>
<p><img src="../reference/figures/first_cross.png" align="left" alt="" width="260">
One thing triangle plots can help do is identify early generation
hybrids, such as F1s. An F1 is the offspring of a cross between an
individual from parental population 1 and parental population 2. The
illustration to the left shows the alleles present at the AIMs we have
identified. Because offspring inherit one allele from each parent, we
can see that the F1 has a blue allele and a red allele at each AIM. In
order to build a triangle plot, we first need to calculate hybrid index
and interclass heterozygosity. Hybrid index is calculated by counting
the number of alleles inherited from parental population 2, and dividing
by the total number of alleles. This gives us a value that can range
between 0 and 1. In this case, the F1 has four alleles from parental
population 2 (blue), and eight total alleles. This gives a hybrid index
of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mfrac><mn>4</mn><mn>8</mn></mfrac><mo>=</mo><mn>0.5</mn></mrow><annotation encoding="application/x-tex">\frac{4}{8}=0.5</annotation></semantics></math>.
Interclass heterozygosity is calculated by counting the number of AIMs
that have an allele from both parental populations, and dividing by
total number of AIMs. In this case there are four AIMs that have an
allele from both parental populations, and four total AIMs, meaning the
interclass heterozygosity is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mfrac><mn>4</mn><mn>4</mn></mfrac><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">\frac{4}{4}=1</annotation></semantics></math>.</p>
<p><img src="../reference/figures/second_cross.png" align="left" alt="" width="260">
Let’s consider another type of cross. The illustration to the left shows
the offspring of a cross between two F1s, called an F2. We can calculate
hybrid index and interclass heterozygosity for the F2 manually as we did
above. In the F2, we see there are four alleles from parental population
2 (blue), so hybrid index is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mfrac><mn>4</mn><mn>8</mn></mfrac><mo>=</mo><mn>0.5</mn></mrow><annotation encoding="application/x-tex">\frac{4}{8}=0.5</annotation></semantics></math>.
In contrast to the F1 though, the F2 only has two AIMs with alleles from
both parental populations, so interclass heterozygosity is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mfrac><mn>2</mn><mn>4</mn></mfrac><mo>=</mo><mn>0.5</mn></mrow><annotation encoding="application/x-tex">\frac{2}{4}=0.5</annotation></semantics></math>.
These are the expected values for an F2, as we will see in the next
section. <span class="math display">$$\\[0.1in]$$</span></p>
</div>
<div class="section level3">
<h3 id="punnett-squares">Punnett Squares
<img src="../reference/figures/punnett_squares.png" align="right" alt="" width="280"><a class="anchor" aria-label="anchor" href="#punnett-squares"></a>
</h3>
<p>Another way to think about the expected genotypes in hybrid offspring
is with Punnett squares. The Punnett square on the top shows the
possible genotypes at an AIM in an F1. Because the parents are
homozygous for different alleles, the only possible genotype is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>12</mn></msub><annotation encoding="application/x-tex">p_{12}</annotation></semantics></math>.
Since the parents have the same genotype at all AIMs, F1s will have the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>12</mn></msub><annotation encoding="application/x-tex">p_{12}</annotation></semantics></math>
genotype at all AIMs.</p>
<p>Now let’s look at a Punnett square for a cross between two F1s. We
see that for any given AIM, there are three possible genotypes in an F2.
There is a 25% chance of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>11</mn></msub><annotation encoding="application/x-tex">p_{11}</annotation></semantics></math>
genotype, a 50% chance of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>12</mn></msub><annotation encoding="application/x-tex">p_{12}</annotation></semantics></math>
genotype, and a 25% chance of the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>22</mn></msub><annotation encoding="application/x-tex">p_{22}</annotation></semantics></math>
genotype. At any given AIM, the F2 can only have one genotype, but if we
had a dataset of 100 AIMs, we would expect that 25 would display the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>11</mn></msub><annotation encoding="application/x-tex">p_{11}</annotation></semantics></math>
genotype, 50 would display the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>12</mn></msub><annotation encoding="application/x-tex">p_{12}</annotation></semantics></math>
genotype, and 25 would display
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>22</mn></msub><annotation encoding="application/x-tex">p_{22}</annotation></semantics></math>
genotype.</p>
</div>
<div class="section level3">
<h3 id="borders-of-a-triangle-plot">Borders of a Triangle Plot<a class="anchor" aria-label="anchor" href="#borders-of-a-triangle-plot"></a>
</h3>
<p><img src="../reference/figures/labeled_tri_plot.png" align="left" alt="" width="400">
So, after seeing how hybrid index and interclass heterozygosity are
calculated, let’s see where individuals appear on a triangle plot. On
the plot to the left, we can see that F1s and F2s both have a hybrid
index of 0.5, but can be distinguished because F1s have an interclass
heterozygosity of 1. BC on the plot stands for “backcross”. A backcross
is when a hybrid crosses with parental individual. BC1 indicates the
offspring of a cross between an F1 and a member of parental population 1
(P1). BC2 indicates a cross between BC1 and P1, and so forth. The solid
black lines forming a triangle indicate the possible space on the plot.
All individuals must be either on the lines or within the triangle. This
is because hybrid index and interclass heterozygosity are not fully
independent of each other. To see why, let’s consider the example
below.</p>
<p><img src="../reference/figures/possible_space.png"></p>
<p>The triangle plots above show an invalid (left) and valid (right)
combination of hybrid index and interclass heterozygosity. Let’s say we
used 100 AIMs to calculate hybrid index and interclass heterozygosity.
To see why the gray point with the red X next to it is not possible,
let’s consider the implications of its interclass heterozygosity. An
interclass heterozygosity of 0.6 means that of the 100 AIMs, 60 of them
are heterozygous, or have a genotype of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>12</mn></msub><annotation encoding="application/x-tex">p_{12}</annotation></semantics></math>.
So, we know there are a minimum of 30
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>2</mn></msub><annotation encoding="application/x-tex">p_2</annotation></semantics></math>
alleles. Thus, even if the genotype at every other AIM is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>11</mn></msub><annotation encoding="application/x-tex">p_{11}</annotation></semantics></math>,
there are still 30
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>2</mn></msub><annotation encoding="application/x-tex">p_2</annotation></semantics></math>
alleles, meaning the hybrid index is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mfrac><mn>30</mn><mn>100</mn></mfrac><mo>=</mo><mn>0.3</mn></mrow><annotation encoding="application/x-tex">\frac{30}{100}=0.3</annotation></semantics></math>.
So, the combination of a hybrid index of 0.25 and interclass
heterozygosity of 0.6 is impossible, because a hybrid index of 0.25
means there can be at most 25
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>2</mn></msub><annotation encoding="application/x-tex">p_2</annotation></semantics></math>
alleles but an interclass heterozygosity of 0.6 means there must be at
least 30
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mn>2</mn></msub><annotation encoding="application/x-tex">p_2</annotation></semantics></math>
alleles.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Ben Wiens.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
